web/packages/design/src/DataTable/StyledTable.tsx

```diff
@@ -89,6 +97,23 @@ export const StyledTable = styled.table(
   borderRadius
 );
 
+// When `border-collapse: collapse` is set on a table element, Safari incorrectly renders row border with alpha channel.
+// It looks like the collapsed border was rendered twice, that is, opacity 0.07 looks like opacity 0.14 (this is more visible
+// on dark theme).
+// Sometimes, there is also an artifact visible after hovering the rows - some of them have correct border color, some not.
+//
+// `getSolidRowBorderColor` is a workaround. The colors below were created by lightening or darkening the table background by
+// the value of theme.colors.spotBackground[0]
+function getSolidRowBorderColor(theme) {
+  const alpha = decomposeColor(theme.colors.spotBackground[0]).values[3];
+  if (theme.name === 'dark') {
```

Comment:
Regarding what I said on #dev-frontend about not using theme name conditionals, could you instead use the `emphasize` function from `colorManipulator.js`? I'm not sure if it achieves exactly what you want here, you might need to reimplement it reversing the logic if needed.

Reply:
This is awesome, thank you! `empasize` works without any changes.